### DISCUSSION TITLE
Import Bibles, Commentaries, Reference Works, etc, from files

### DISCUSSION THREAD

**Discussion by theophiluso7** [ezra-bible-app/ezra-bible-app#360] [Q&A] [answered]
Tobias, I am asking you to consider putting an importer into Ezra Bible app. I hope that for Bibles & maybe commentaries, that importing them would result in the same mechanical functionality as Bibles downloaded from the repositories, and not simply copy and pasted ebook files. The resources available are very limited and largely extremely outdated. _Having to buy resources entirely defeats the purpose_ of Ezra for other Christians who have little or no money. I know there are reasons why you might not want to do this. Ezra is the only modern free Bible app I am aware of. The problem is the app is in part, only as good as the available resources. Faithlife has deliberately not included an any serious way to import our own books and have them not be broken. The same applies partly to Accordance. These companies are not incentivized to let us use our own books.

@tobias-klein 
This is centrally important. I'm hoping you will consider it.

**Comment by tobias-klein** [comment, marked as answer]
Hi @theophiluso7,

Thank you for your suggestion! At the moment, Ezra Bible App is a SWORD frontend like any other SWORD-based tool and it relies on the SWORD module format. I realize that many of the resources in the SWORD module repos are rather old. However, what problem would an "importer" solve? And if it would exist it would technically only solve the problem for one user, right? Wouldn't it make more sense to invest our energy into getting more and better centrally available module resources? There is a module team over at the CrossWire project, which works on making more and updated resources available. See [this Wiki page](http://wiki.crosswire.org/DevTools:Modules), which gives some starting points on how to create new modules. Could you imagine to join the efforts?

For module translations, people like us need to work with the copyright holders / publishers to see whether more translations can be provided as SWORD modules - either free or payed. One nice recent example is the Lockman Foundation who have started to provide the NASB and also their Spanish translations in their own Repo of free SWORD modules.

Which particular resources were you thinking of when it comes to "import"?

**Reply by tobias-klein** [reply]
@theophiluso7 Still curious about your feedback to my response.

**Reply by theophiluso7** [reply]
Tobias, I'm referring to PDFs, Epubs, or and possibly other ebooks. I own Logos, and really don't need Ezra. However, I'm going to be straight with you. Ezra is largely useless if it requires any substantial amount of money. It would solve the problem for MANY individual users. The fact is that Bible software is extremely expensive and resources aren't affordable. I do not believe that these resources should be under an unpassable paywall.

**Reply by tobias-klein** [reply]
@theophiluso7 I understand your argument. However, it's not the primary mission of Ezra Bible App to manage existing resources.
The purpose/mission of Ezra Bible App is to have a study software that allows you to manage your own work connected to the Bible (your own commentary / notes, tags, etc.). Of course this would be better if more modern translations would be available - we can work on that together with the publishers.

Since Ezra works with SWORD modules a generic "importer" would essentially be a piece of software that turns some kind of input file into a SWORD module. One project that looks like it is going into that direction is https://github.com/schierlm/BibleMultiConverter. Maybe you should check that out?

**Reply by theophiluso7** [reply]
Thank you. I'll get back to you about that as soon as possible. I appreciate the help.

**Reply by theophiluso7** [reply]
Tobias, I have not had luck with BMC. I don't know how to use it. I might be doing things wrong. Is there a clear set of simple instructions?

**Reply by tobias-klein** [reply]
@theophiluso7 The BibleMultiConverter guys can probably help better - I have never used it either so far. They say in their README that you should [open an issue](https://github.com/schierlm/BibleMultiConverter/issues/new) with your question if anything is unclear.
